packages/cli/src/ui/components/InputPrompt.test.tsx

```diff
@@ -1364,17 +1367,37 @@ describe('InputPrompt', () => {
 
       await act(async () => {
         stdin.write('\x1B');
-        await waitFor(() => {
-          expect(onEscapePromptChange).toHaveBeenCalledWith(false);
-        });
+        vi.advanceTimersByTime(100);
+
+        expect(onEscapePromptChange).toHaveBeenCalledWith(false);
       });
 
       await act(async () => {
         stdin.write('\x1B');
-        await waitFor(() => {
-          expect(props.buffer.setText).toHaveBeenCalledWith('');
-          expect(mockCommandCompletion.resetCompletionState).toHaveBeenCalled();
-        });
+        vi.advanceTimersByTime(100);
+
+        expect(props.buffer.setText).toHaveBeenCalledWith('');
+        expect(mockCommandCompletion.resetCompletionState).toHaveBeenCalled();
+      });
+      unmount();
+    });
+
+    it('should clear buffer on double ESC', async () => {
```

Comment:
can you add a test for this case with some mouse scroll commands tossed in the middle? I'm able to get the vscode terminal to ignore escapes if scrolling rapidly at the same time with these changes.